redis/cluster.py

```diff
@@ -17,16 +17,17 @@
     ClusterCrossSlotError,
     ClusterDownError,
     ClusterError,
+    ConnectionError,
     DataError,
     MasterDownError,
     MovedError,
     RedisClusterException,
     RedisError,
     ResponseError,
     SlotNotCoveredError,
-    TimeoutError,
-    TryAgainError,
 )
+from redis.exceptions import TimeoutError as RedisTimeoutError
+from redis.exceptions import TryAgainError
```

Comment:
isort forces to take out the  TryAgainError as well..

Reply:
TimeoutError is defined in exceptions.py, and inherited from RedisError.  Either we should use the pure TimeoutError declared there (rather than RedisTimeoutError), or change that error everywhere to be named RedisTimeoutError.

If you want to keep the dedicated (and different) RedisTimeoutError as if it's not a stock error, then we should add a ClusterTimeoutError class and you can capture that. However - I don't think this is the case.

Reply:
I changed it to catch also socket.timeout (the default TimeoutError) because in one of the tests I did I thought connection.py thrown this error and not the redis.exceptions.TimeoutError. However, I go over again on connection.py code and I can't find a place where it doesn't wrap it with redis's timeoutError. I tried to reproduce it to see if I get again socket.timeout error but I was not able to find it again. 
I reverted it and discard the default timeout error, so we only catch redis's timeoutError in the cluster's code. I tested it again couple more time with live failovers and with short timeout periods and the client handles it properly.
So, as I said - I rolled back the timeout error import etc., please disregard it.